plugins/techdocs/src/reader/components/useReaderState.ts

```diff
@@ -216,11 +221,18 @@ export function useReaderState(
   namespace: string,
   name: string,
   path: string,
-): { state: ContentStateTypes; content?: string; errorMessage?: string } {
+): {
+  state: ContentStateTypes;
+  contentReload: () => void;
```

Comment:
The `contentReload` strikes back 😄 (https://github.com/backstage/backstage/pull/5975#discussion_r651879164). While we still don't produce side effect in the reducer, we now have two different places (`useReaderState` and `<Reader>`) in which we manually trigger a change by circumventing the state machine.

I'm suggesting to replace the `useAsyncRetry` with an `useEffect` that only fires when the `BUILD_READY_RELOAD` state is being reached. Once the `useEffect` has finished fetching, it should dispatch `BUILD_READY` (or similar). That means the interface to produce side effects is by dispatching states, but the hook itself takes care of return the expected data to the consumer.

Reply:
I think the main issue comes from the fact that I tried to combine two independent processes in the same hook:
1. Download the most recent content of a page regardless of it being fresh or not.
2. Check if the contents in the e.g. S3 bucket are fresh and if not, automatically refresh it, regardless of whether the content was downloaded or not.

With **2.** having the special desire to automatically redo **1.** if-and-only-if there was nothing to show before. And now the user also wants to be able to redo **1.** whenever he likes. But he doesn't want to infer with the sync process of **2.** while doing it.

I also noticed that the combination of the states sometimes behaves a bit weird and that it might be better to decouple the header (is there a sync and what is their outcome) from the content (is it loading and can it be displayed) by putting **1.** and **2.** into individual hooks and move this special desire of **2.** up into the `<Reader/>` component.

Reply:
But this change (adding the refresh button when ready) is also not connected to the event-stream functionality. It might be better to just leave it out for now and revisit the case in a different PR.